mappings/net/minecraft/client/texture/NativeImage.mapping

```diff
@@ -1,15 +1,57 @@
 CLASS net/minecraft/class_1011 net/minecraft/client/texture/NativeImage
 	CLASS class_1012 Format
 		FIELD field_4993 pixelDataFormat I
-		FIELD field_4994 bytesPerPixel I
+		FIELD field_4994 totalChannels I
+		FIELD field_4996 writeable Z
+		FIELD field_4999 alpha Z
+		FIELD field_5000 luminance Z
+		FIELD field_5003 blue Z
+		FIELD field_5004 green Z
+		FIELD field_5005 red Z
+		FIELD field_5006 alphaChannelOffset I
+		FIELD field_5007 luminanceChannelOffset I
+		FIELD field_5008 blueOffset I
+		FIELD field_5009 greenOffset I
+		FIELD field_5010 redOffset I
+		METHOD <init> (Ljava/lang/String;IIIZZZZZIIIIIZ)V
+			ARG 3 size
+			ARG 4 glFormat
+			ARG 5 red
+			ARG 6 green
+			ARG 7 blue
+			ARG 8 luminance
+			ARG 9 alpha
+			ARG 10 redOffset
+			ARG 11 greenOffset
+			ARG 12 blueOffset
+			ARG 13 luminanceOffset
+			ARG 14 alphaOffset
+			ARG 15 writeable
+		METHOD method_4329 hasAlphaChannel ()Z
```

Comment:
Does it only have an alpha channel if it has translucent pixels? If so, maybe `hasTranslucency`?

Reply:
alpha = translucency

Alpha is the opacity channel in almost every format I know. The only one that _doesn't_ have alpha, is gifs. Those use an image-wide mapping to assign a specific colour to transparent pixels and don't support translucency at all.